src/Assert.php

```diff
@@ -840,6 +841,8 @@ public static function notFalse($value, $message = '')
     }
 
     /**
+     * @psalm-pure
```

Comment:
anything using `filter_var` cannot be pure, as it implicitly causes `__toString`